### PR TITLE
support check suppression via CFN Metadata

### LINKED ISSUE #1931: Allow suppression to be done in Metadata properties of the AWS CloudFormation template and/or resources
Allow suppression to be done in Metadata properties of the template and/or resources

### LINKED ISSUE #2194: CKV_AWS_116: DLQ applies only to async invocations
**Describe the issue**
CKV_AWS_116: "Ensure that AWS Lambda function is configured for a Dead Letter Queue(DLQ)"
DLQ applies only to synchronous Lambda invocations, so if you have a REST API for example it won't have any effect and thus is unnecessary. The rule fires up for all Lambdas.

I understand that unfortunately in a general case it is impossible to figure out from a CloudFormation template with a Lambda function whether it is synchronous or asynchronous, because you can call any function in either way with AWS CLI or API, without any event mapping.

### PR DESCRIPTION
**Pull request by gruebel**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Fixes #2194
Fixes #1931

Extended the graph to also store the CloudFormation `Metadata` and added the possibility to actually add the skip comment that section. Added an example to the docs.

### CODE REVIEW

File: checkov/cloudformation/runner.py
Comment (Saarett) [inline checkov/cloudformation/runner.py:115]: We have we added this step?

**Review by Saarett — APPROVED** [review]
Very nice addition! Looks good to me, added 1 comment in the code.
Please add a test for metadata in json files.
Please wait for another approval :)

File: docs/2.Basics/Suppressing and Skipping Policies.md
Comment (schosterbarak) [inline docs/2.Basics/Suppressing and Skipping Policies.md:136]: show example bash run

File: docs/2.Basics/Suppressing and Skipping Policies.md
Comment (schosterbarak) [inline docs/2.Basics/Suppressing and Skipping Policies.md:152]: show the printed cli output when scanning this manifest to show that a policy is being skipped

File: docs/2.Basics/Suppressing and Skipping Policies.md
Comment (schosterbarak) [inline docs/2.Basics/Suppressing and Skipping Policies.md:118]: what is L1?

File: checkov/cloudformation/context_parser.py
Comment (nimrodkor) [inline checkov/cloudformation/context_parser.py:144]: @gruebel Let's elevate to `info` / `warning` please, this is highly unexpected! Also add more identifiable data to the log to help the suppressor identify the problematic suppression

File: checkov/cloudformation/graph_builder/graph_to_definitions.py
Comment (nimrodkor) [inline checkov/cloudformation/graph_builder/graph_to_definitions.py:28]: @gruebel Why not set it inside the `definition = {...`?

**Review by nimrodkor — COMMENTED** [review]
Very nice!

File: checkov/cloudformation/graph_builder/graph_to_definitions.py
Comment (gruebel) [reply checkov/cloudformation/graph_builder/graph_to_definitions.py:28]: because it is optional, not every resource has a `Metadata` section. It would be misleading, if I add it with an empty `dict` or `None` value

File: checkov/cloudformation/context_parser.py
Comment (gruebel) [reply checkov/cloudformation/context_parser.py:144]: makes sense, let's go with `warning` and if users complain being to noisy, then we can reduce it later 😄

File: docs/2.Basics/Suppressing and Skipping Policies.md
Comment (gruebel) [reply docs/2.Basics/Suppressing and Skipping Policies.md:118]: this a term used in CDK docs, it stands for Layer 1 or also called Level 1. These are the low level CloudFormation resources. There also L2 and even L3 constructs, which are abstractions of the low level CloudFormation resources.

File: checkov/cloudformation/runner.py
Comment (gruebel) [reply checkov/cloudformation/runner.py:114]: removed

File: docs/2.Basics/Suppressing and Skipping Policies.md
Comment (gruebel) [reply docs/2.Basics/Suppressing and Skipping Policies.md]: please have a look again

**Review by schosterbarak — APPROVED** [review]

**Pull request merged**
